### PR TITLE
Error handling while create folder in media manager

### PR DESCRIPTION
**Pull request by Kaushik1216**
Pull Request for Issue #39263

### Summary of Changes
Closing Create folder Modal On error while creating folder

### Testing Instructions
Create a folder as mention in issue like folder with name ( joomla..... )

### Actual result BEFORE applying this Pull Request

The create modal not closed to show error in case of error occuring while creating folder

### Expected result AFTER applying this Pull Request

create modal closed to show error

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by dgrammatiko** [comment]
@Kaushik1216 around 10 lines before the line you've added there are couple of todos that should point you to the needed changes.
Essentially what your code is doing is closing the modal always although we should keep the modal open when there's an error, mark the field as invalid and also render (inside the modal) the error message. In short you're changing the workflow here and introducing more clicks to complete a task if there's an error.

**Comment by Kaushik1216** [comment]
@dgrammatiko yes you are right ,so what can be best way to solve this issue .One way I think is ,when undesire folder name come,input field should become red and some text like invalid folder name above input field can be shown.

**Comment by Kaushik1216** [comment]
@dgrammatiko  Please check this changes this  will do as follow .

https://user-images.githubusercontent.com/100058812/218261098-3c7dbf15-f020-4605-8e28-782acf19b5ef.mp4

**Comment by dgrammatiko** [comment]
> Please check this changes this will do as follow .

There are many more cases that Joomla (the PHP side) will not accept a name as valid, so somehow you need to check for all these.

**Comment by Kaushik1216** [comment]
> > Please check this changes this will do as follow .
> 
> There are many more cases that Joomla (the PHP side) will not accept a name as valid, so somehow you need to check for all these.

Ok ,then please provide me those cases I will do similiar things for all cases

**Comment by dgrammatiko** [comment]
> Ok ,then please provide me those cases I will do similiar things for all cases

Dude if I have to do the research then I could as well do the PR...

**Comment by Kaushik1216** [comment]
> > Ok ,then please provide me those cases I will do similiar things for all cases
> 
> Dude if I have to do the research then I could as well do the PR...

Hmm, Ok then I will do research and find some more constraint

**Comment by Kaushik1216** [comment]
Error show if fiolder name already exist in directory as folder or file name

error show as follows

1createfolderfileexisterror

**Comment by Kaushik1216** [comment]
@dgrammatiko  Is I am doing right for multiple case handling ?

**Comment by Kaushik1216** [comment]
AS I notice php side only accept folder name which contain only Alphanumeric,underscore(_) ,period ( . ) and hyphen ( - ).
if you include any other charachater other than this php remove all other character from folder name .
Problem is that if you have a folder name for example ss and you attempt to try create folder with name s@s then you not get any error and folder also not created because php remove @ from folder name and ss result new folder name which is already present their and bad thing is that user also get a success msg Folder created .

**Comment by Kaushik1216** [comment]
the only one error That I think should come from php side is when user not install INTL and ICONV php extension and writing non ascii character in folder name should we close modal for this error ?

File: administrator/components/com_media/resources/scripts/components/modals/create-folder-modal.vue [hunk 1]
```diff
@@ -65,11 +88,37 @@ export default {
       folder: '',
     };
   },
+  computed: {
+    /* Get the contents of the currently selected directory */
+    items() {
+      // eslint-disable-next-line vue/no-side-effects-in-computed-properties
+      const directories = this.$store.getters.getSelectedDirectoryDirectories
```
Comment (dgrammatiko) [inline administrator/components/com_media/resources/scripts/components/modals/create-folder-modal.vue:95]: If you have to add an eslint flag to overcome the no side effects that’s a very good indication that the code is problematic! Basically DO NOT mutate items is the clue here!

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

**Comment by dgrammatiko** [comment]
Please rebase this to 4.3!

**Comment by dgrammatiko** [comment]
Lastly the media manager has the concept of adapters, ie supporting different online storage solutions (Amazon s3, Dropbox, etc). Therefore the checks here should cover all these options…

File: administrator/components/com_media/resources/scripts/components/modals/create-folder-modal.vue (hunk 1 above)
Comment (Kaushik1216) [reply administrator/components/com_media/resources/scripts/components/modals/create-folder-modal.vue:95]: got it

**Comment by Kaushik1216** [comment]
> Please rebase this to 4.3!

I think rebaseing this pr with 4.3 will increse work more that also occure in my no search result pr .
So I am creating new Pr for 4.3 branch .Should I closed it then ?

**Comment by Kaushik1216** [comment]
closing by creating another Pr #39878 please refer this

**Pull request closed without merging**